src/entry/data-provider/index.ts

```diff
@@ -31,6 +35,32 @@ export const dataProvider = async (request: DataProviderPayload): Promise<DataPr
 
   const trackingBranch = await getTrackingBranchName(groupToken, projectId, defaultBranch);
 
+  const backfillData: BackfillData = {
+    builds: [],
+    deployments: [],
+    metrics: {
+      mrCycleTime: 0,
+      openMergeRequestsCount: 0,
+    },
+  };
+
+  try {
+    const {
+      builds,
+      deployments,
+      metrics: { mrCycleTime, openMergeRequestsCount },
+    } = await getBackfillData('groupToken', projectId, projectName, trackingBranch);
+
+    backfillData.builds = builds;
+    backfillData.deployments = deployments;
+    backfillData.metrics.mrCycleTime = mrCycleTime;
+    backfillData.metrics.openMergeRequestsCount = openMergeRequestsCount;
+  } catch (err) {
+    const invocationErrorOptions = { backoffTimeInSeconds: 0 };
+
+    return new ForgeInvocationErrorResponse(err.statusText, err.status, invocationErrorOptions).build();
```

Comment:
Does the Gitlab fetchfn throw errors with this status and statusText already filled out when we get non 200 responses?

Reply:
Yes, we throw this error with filled status-related fields from the client. We will also add the check for error instances and return 500 in case of non-API errors.